runtime/src/dpe_crypto.rs

```diff
@@ -286,14 +286,15 @@ impl<'a> Crypto for DpeCrypto<'a> {
                 // note: the output point must be kept secret since it is derived from the private key,
                 // so as long as that output is kept secret and not released outside of Caliptra,
                 // it is safe to use it as key material.
-                let (_, hmac_seed) = Self::derive_key_pair(self, algs, cdi, label, info)?;
+                let (_, mut hmac_seed) = Self::derive_key_pair(self, algs, cdi, label, info)?;
 
                 // create ikm to the hmac kdf by hashing the seed entropy from the pub key
                 // this is more secure than directly using the pub key components in the hmac
                 // kdf since the distribution of the pub key is taken from a set of discrete points
                 let mut hasher = Self::hash_initialize(self, algs)?;
```

Comment:
Could we call this before `derive_key_pair`, so an error path won't leave `hmac_seed` around? Probably less likely that `hasher.update()` would fail.

Reply:
Not sure if we can since then we would have multiple mutable borrows.

Reply:
I think we should use ZeroizeOnDrop to avoid this issue.

Reply:
Either way, I don't think its possible since we would have something like 
<img width="875" alt="image" src="https://github.com/chipsalliance/caliptra-sw/assets/105384008/f8085ebd-5b5f-4fd4-b73a-f744f3f1ee4a">
, unless I am understanding wrong.

Reply:
Oh, I agree we want initialize hasher first. But if we do ZeroizeOnDrop then even if `hash_initialize` fails, hmac_seed will still get dropped.

Reply:
With https://github.com/chipsalliance/caliptra-dpe/pull/271, hmac_seed will be zeroized when dropped so this should also be fixed.